src/main/java/com/microsoftopentechnologies/windowsazurestorage/WAStorageClient.java

```diff
@@ -320,9 +310,12 @@ public static int upload(Run<?, ?> run, Launcher launcher, TaskListener listener
 			// Remove the workspace bit of this path
 			URI srcURI = workspaceURI.relativize(src.toURI());
 
-			InputStream inputStream = src.read();
-			String md5hex = DigestUtils.md5Hex(inputStream);
-			long sizeInBytes = src.length();
+			String md5hex;
+			long sizeInBytes;
+			try (InputStream inputStream = src.read()){
+					md5hex = DigestUtils.md5Hex(inputStream);
```

Comment:
Seems inefficient to open and read the stream twice (once to get the md5hex and once to upload) so I looked around and found a more elegant solution (assuming it works): http://stackoverflow.com/questions/304268/getting-a-files-md5-checksum-in-java

In this case, the inner 'upload' function could just return the hash